src/core/providers/ogr/qgsogrfeatureiterator.cpp

```diff
@@ -815,7 +815,7 @@ QgsOgrFeatureSource::QgsOgrFeatureSource( const QgsOgrProvider *p )
   , mShareSameDatasetAmongLayers( p->mShareSameDatasetAmongLayers )
   , mLayerName( p->layerName() )
   , mLayerIndex( p->layerIndex() )
-  , mSubsetString( p->mSubsetString )
+  , mSubsetString( p->mSubsetString.isEmpty() ? QString() : QgsOgrProviderUtils::cleanSubsetString( p->mSubsetString ) )
```

Comment:
Wouldn't it be cleaner to make QgsOgrProviderUtils::cleanSubsetString return an empty string if the input string is empty?

Reply:
good point. It actually does. Code simplified to ``, mSubsetString( QgsOgrProviderUtils::cleanSubsetString( p->mSubsetString ) )``